### ISSUE TITLE
Proposal: Move node-private to a separate GitHub org

### ISSUE THREAD

**Issue by jasnell** [nodejs/TSC#320]
This is not a proposal that can be made with a pull request. I would like to propose that a separate GitHub organization be established to host the nodejs/node-private repository and security vulnerabilities details. Access to this repository would be limited to the @nodejs/security team whose membership would continue to be under the oversight of the @nodejs/ctc and @nodejs/tsc (or the combined version of those assuming that moves forward.

@nodejs/ctc and @nodejs/tsc ... please weigh in on this.

**Comment by mcollina** [comment]
LGTM, it solves **a lot** of problems.

**Comment by MylesBorins** [comment]
+1

We do need to find a similarly elegant solution for CI as well.

Would this affect any of the permissions currently in ci?

On Aug 23, 2017 7:52 AM, "Matteo Collina" <notifications@github.com> wrote:

LGTM, it solves *a lot* of problems.

—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/nodejs/TSC/issues/320#issuecomment-324361104>, or mute
the thread

**Comment by jasnell** [comment]
While emoji feedback is good, I would appreciate if @nodejs/ctc and @nodejs/tsc members would indicate their support explicitly with a comment so it's clearer. Thank you!

**Comment by addaleax** [comment]
I’m completely in favour.

**Comment by jasnell** [comment]
For CI, I would leave that up to @nodejs/build to figure out. Yes, there would be an impact on CI configuration.

**Comment by evanlucas** [comment]
I am in favor.

**Comment by mhdawson** [comment]
I'm +1 we'll just need to work through any of the CI/other problems.

**Comment by cjihrig** [comment]
+1

**Comment by gibfahn** [comment]
>We do need to find a similarly elegant solution for CI as well.
>Would this affect any of the permissions currently in ci?

It will require an update, but it should be pretty straightforward.

**Comment by sam-github** [comment]
+1, needed for the node fork where private security patches are prepared prior to publication. The issues around access to the security issues themselves would be better handled by using a issue tracker designed for security, IMO, such as HackerOne, but that isn't a reason to not create a private org immediately and move these two repos.

**Comment by targos** [comment]
+1

FWIW nodejs-private and nodejs-security both seem available for the name of the org.

**Comment by jasnell** [comment]
I have just created https://github.com/nodejs-private and will transfer ownership to the Foundation assuming this goes through. The org account will need to be updated to a paid account in order to have the private repos. /cc @hackygolucky

**Comment by bnoordhuis** [comment]
+1

**Comment by refack** [comment]
@jasnell I wanted to ask that the name would be different, since "node-private" is an internal jargon, and it's meaning is not obvious to outsiders.
But since you already created it, I think it would be simpler to just add a few words in the description field, and that will achieve the same goal, of clarifying what is the purpose of this org.

**Comment by jasnell** [comment]
Orgs can be renamed fairly easily :-) I've added a short description.

**Comment by ofrobots** [comment]
+1

**Comment by bengl** [comment]
Out of curiosity, what are the issues that are solved by moving this to a separate org? I have ideas, but it would be great to have them enumerated here (or elsewhere if there's an existing issue?).

**Comment by addaleax** [comment]
@bengl Two things that I remember having come up repeatedly so far:

- There are too many org admins, or just generally the set of org admins doesn’t match the ideal set of people who should have access to sensitive information
- Having repositories containing sensitive code is a blocker for enabling e.g. Travis CI for the github organization

**Comment by hackygolucky** [comment]
Ping me as soon as it gets merged and I'm happy to take care of this!

**Comment by jasnell** [comment]
@hackygolucky the steps here are simple: I just need a new billing contact for the private org and will add the members of the @nodejs/security team as owners. From there it's just a matter of transferring the node-private repo over which requires CTC / TSC  sign off.

**Comment by targos** [comment]
Can we ask GitHub for free private repositories? That's what we have for the main org.

**Comment by jasnell** [comment]
We will be

**Comment by jasnell** [comment]
I have added all of the current members of @nodejs/security as owners on the new org. It is still not set up to have private repositories so please do not set anything up yet. I have asked @hackygolucky and @mrhinkle to get the process started there. Once everything is set up, we can transfer the `*-private` repos over.

/cc @nodejs/ctc @nodejs/tsc

**Comment by jasnell** [comment]
@hackygolucky ... I have made you the billing contact for the organization. Please let me know as soon as the account is enabled for private repos and I will start the process of moving things over.

@nodejs/ctc and @nodejs/tsc members... please weigh in if you have objections to this proposal. If there are no objections by next Wednesday, I will assume consent and proceed.

**Comment by Fishrock123** [comment]
Can we please do this (I advocated for this to be done well over a year ago), I'm sure GitHub would probably be ok with giving us 2-3 private repos elsewhere.

_(Can I please just have the ability re-enable travis for WGs that bugged me for months to turn it on? That would be great thanks lol.)_

**Comment by jasnell** [comment]
The process is already underway.

**Comment by Fishrock123** [comment]
As a note, can we try to get full repo transfers, issue tracker and all? I forget if that is easy or not but would certainly be ideal.

**Comment by jasnell** [comment]
That's the plan. Once private repos are enabled on the account, we simply need for there to be no objections from the CTC/TSC for us to initiate a transfer. It will take me all of about 1 minute to complete the actual transfer once I'm cleared to do so.

**Comment by jasnell** [comment]
(the biggest thing we're waiting on would be the account be approved to have private repos)

**Comment by jasnell** [comment]
Another thing we still need to do,btw, is audit the members of the @nodejs/security team to see if we still need that many people on it, but that's a separate issue.

**Comment by gibfahn** [comment]
>(Can I please just have the ability re-enable travis for WGs that bugged me for months to turn it on? That would be great thanks lol.)

Are there currently any working groups who are bugging you to use Travis?

**Comment by mcollina** [comment]
readable-stream is still using Travis, ad we need to finish migrating everything (sigh, time). But Travis is not working well for us anyway.

**Comment by Fishrock123** [comment]
> Are there currently any working groups who are bugging you to use Travis?

No, that fate was resigned to like a year ago. :/

**Comment by jasnell** [comment]
Ping @hackygolucky or @mrhinkle ... can either of you provide an update on whether the new org has been set up for private repos yet?

**Comment by jasnell** [comment]
@MylesBorins ... will need you to follow up on this with Mark to make sure it gets done. We need to get the security repos moved.

**Comment by MylesBorins** [comment]
will do thanks for heads up

**Comment by MylesBorins** [comment]
@jasnell I just checked new org is set up for unlimited private repositories.

**Comment by MylesBorins** [comment]
@nodejs/security is there someone that wants to take point on moving *-private repos over to the new org?

**Comment by jasnell** [comment]
I can move them.

**Comment by jasnell** [comment]
Moved node-private and http-parser-private. The nodejs.org-private was not moved.

**Comment by jasnell** [comment]
build-private is also still there. The @nodejs/website and @nodejs/build teams need to decide whether they need those moved also.

**Comment by targos** [comment]
I think we should enforce 2FA in the private org if all members have it enabled.

**Comment by fhemberger** [comment]
@jasnell I don't think `build-private` is relevant for the website, as long as `build` is around. We just need to adjust nginx rewrites occasionally.

BTW: I'm **so** happy about this decision, now we can finally add translation workflow tools to nodejs.org, which will make things incredibly easier for us.

**Comment by jasnell** [comment]
@targos ... I can do that but it would remove nodejs-ci bot account from the org. I think that is ok for now. We'll need to revisit it tho. Ping @nodejs/build

**Comment by jasnell** [comment]
2FA is now required for the new repo.

**Comment by jasnell** [comment]
Closing this issue now that the org is set up and the repos have been moved.

**Comment by mcollina** [comment]
I do not have access.

**Comment by jasnell** [comment]
You're not on the nodejs/security team? We should fix that. Currently, only members of the nodejs/security team + @mrhinkle and @hackygolucky have access.

**Comment by joaocgreis** [comment]
> I can do that but it would remove nodejs-ci bot account from the org. I think that is ok for now. 

That bot is only used for the temporaries now, so removing is ok. It was used before for security releases, we can revisit when we need to test security releases again.
